packages/aws-cdk-lib/aws-stepfunctions/lib/states/state.ts

```diff
@@ -411,11 +427,9 @@ export abstract class State extends Construct implements IChainable {
    * Render map iterator in ASL JSON format
    */
   protected renderIterator(): any {
-    if (!this.iteration) {
-      throw new Error('Iterator must not be undefined !');
-    }
+    if (!this.iteration) return undefined;
```

Comment:
Is it safe to remove the error here? Maybe we could add an additional check for `!this.processor` so that we can still throw the error if neither are set.

Reply:
I removed validation here to handle it in [`validateState`](https://github.com/aws/aws-cdk/pull/27913/files#diff-cca37caeb11e88e3ed977617a93def71dc433b7b12428ab0c7a9d6e964d76e80L194).
I think it's clearer to have it centralized and we shouldn't miss coverage unless I'm missing something.